torch/distributed/__init__.py

```diff
@@ -54,6 +56,32 @@ def is_available() -> bool:
         _make_nccl_premul_sum,
     )
 
+    def breakpoint(rank: int = 0):
+        """
+        Set a breakpoint, but only on a single rank.  All other ranks will wait for you to be
+        done with the breakpoint before continuing.  This calls ``breakpoint()`` under the
+        hood, so you can customize it using the normal facilities, e.g., ``PYTHONBREAKPOINT``
+        environment variable.
+
+        Args:
+            rank (int): Which rank to break on.  Default: ``0``
```

Comment:
Is this mostly just for debugging unit tests? I'm guessing this won't work for real jobs that are submitted through schedulers like slurm/kubernetes.

Seems like this should be under `https://github.com/pytorch/pytorch/tree/main/torch/testing/_internal/distributed` and probably not a public API.

Reply:
Yeah, unit tests, and when I'm running single node stuff triggered manually with torchrun or something.

I decided to put it in torch.distributed because I wanted a relatively short way to reference it when I'm inserting it for debugging purposes. If the module path is really long it's a pain too!